### PR TITLE
Add a check for project submodules.

### PR DESCRIPTION
**Pull request by sinetek**
Greetings.
I have added this check in the CMakeList that warns users if the submodules are missing. It might be useful.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,6 +77,27 @@ endmacro()
...
+  find_package(Git)
+  if(GIT_FOUND)
+    function (check_submodule relative_path)
+      execute_process(COMMAND git rev-parse "HEAD" WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${relative_path} OUTPUT_VARIABLE localHead)
+      execute_process(COMMAND git rev-parse "HEAD:${relative_path}" WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} OUTPUT_VARIABLE checkedHead)
+      string(COMPARE EQUAL "${localHead}" "${checkedHead}" upToDate)
+      if (upToDate)
+        message(STATUS "Submodule '${relative_path}' is up-to-date")
+      else()
+        message(FATAL_ERROR "Submodule '${relative_path}' is not up-to-date. Please update all submodules with\ngit submodule update --init --force\nor run cmake with -DMANUAL_SUBMODULES=1\n")
```
Comment (abitmore) [inline CMakeLists.txt:91]: We usually run `git submodule update --init --recursive` since there are other submodules inside the `fc` submodule.

**Review by abitmore — CHANGES_REQUESTED** [review]
Thanks. Please rebase to `develop` branch and force-push.

File: CMakeLists.txt (hunk 1 above)
Comment (sinetek) [reply CMakeLists.txt:91]: Sure thing.

**Comment by sinetek** [comment]
I applied the patch to develop, but now there's 77 modifications. I'm confused, how do I fix this?

**Pull request closed without merging**